snapshots/windows/windows.go

```diff
@@ -251,25 +251,36 @@ func (s *snapshotter) Remove(ctx context.Context, key string) error {
 	renamedID := "rm-" + id
 	renamed := s.getSnapshotDir(renamedID)
 	if err := os.Rename(path, renamed); err != nil && !os.IsNotExist(err) {
-		if !os.IsPermission(err) {
+		log.G(ctx).WithError(err).Errorf("rename failure during snapshot %s cleanup", id)
+		// Sometimes if there are some open handles to the files (especially VHD)
+		// inside the snapshot directory the rename call will return "access
+		// denied" or "file is being used by another process" errors.  Just
+		// returning that error causes the entire snapshot garbage collection
+		// operation to fail. To avoid that we return failed pre-condition error
+		// here so that snapshot garbage collection can continue and can cleanup
+		// other snapshots.
+		if !(strings.Contains(err.Error(), "being used by another process") || os.IsPermission(err)) {
 			return err
 		}
-		// If permission denied, it's possible that the scratch is still mounted, an
-		// artifact after a hard daemon crash for example. Worth a shot to try deactivating it
-		// before retrying the rename.
-		var (
-			home, layerID = filepath.Split(path)
-			di            = hcsshim.DriverInfo{
-				HomeDir: home,
-			}
-		)
-
-		if deactvateErr := hcsshim.DeactivateLayer(di, layerID); deactvateErr != nil {
-			return errors.Wrapf(err, "failed to deactivate layer following failed rename: %s", deactvateErr)
-		}
+		if os.IsPermission(err) {
```

Comment:
It looks like there's a gap in conditions here; `err` could contain the string "being used by another process" but not be a permission error; that error should probably still be returned.

Reply:
Good catch. Fixed it.